spec/models/contact_details_spec.rb

```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe ContactDetails, type: :model do
+  it { is_expected.to validate_presence_of(:phone_number).on(:update) }
+  it { is_expected.to validate_presence_of(:email_address).on(:update) }
+  it { is_expected.to validate_presence_of(:address).on(:update) }
+
+  # TODO: create relation between ContactDetails and Candidate
```

Comment:
Very good to be thinking about this, but we haven't agreed on an approach for how all this will fit together — please could you remove these two lines for now?